src/ui/viewmodels/MessageBoxViewModel.hh

```diff
@@ -0,0 +1,167 @@
+#ifndef RA_UI_MESSAGEBOX_VIEW_MODEL_H
+#define RA_UI_MESSAGEBOX_VIEW_MODEL_H
+#pragma once
+
+#include "ui/WindowViewModelBase.hh"
+
+namespace ra {
+namespace ui {
+namespace viewmodels {
+
+enum class MessageBoxButtons
+{
+    OK,
+    OKCancel,
+    YesNo,
+    YesNoCancel,
+    RetryCancel,
+};
+
+enum class MessageBoxIcon
+{
+    None,
+    Info,
+    Warning,
+    Error,
+};
+
+class MessageBoxViewModel : public WindowViewModelBase
+{
+public:
+    MessageBoxViewModel() noexcept {}
+
+    MessageBoxViewModel(const std::wstring& sMessage) noexcept
+    {
+        SetMessage(sMessage);
+    }
+
+#pragma push_macro("GetMessage") // windows stole my method name
+#undef GetMessage
+    /// <summary>
+    /// The <see cref="ModelProperty" /> for the message.
+    /// </summary>
+    static StringModelProperty MessageProperty;
+
+    /// <summary>
+    /// Gets the message to display.
+    /// </summary>
+    const std::wstring& GetMessage() const { return GetValue(MessageProperty); }
+
+    /// <summary>
+    /// Sets the message to display.
+    /// </summary>
+    void SetMessage(const std::wstring& sValue) { SetValue(MessageProperty, sValue); }
+#pragma pop_macro("GetMessage")
+
+    /// <summary>
+    /// The <see cref="ModelProperty" /> for the header message.
+    /// </summary>
+    static StringModelProperty HeaderProperty;
+
+    /// <summary>
+    /// Gets the header message to display.
+    /// </summary>
+    const std::wstring& GetHeader() const { return GetValue(HeaderProperty); }
+
+    /// <summary>
+    /// Sets the header message to display, empty string for no header message.
+    /// </summary>
+    void SetHeader(const std::wstring& sValue) { SetValue(HeaderProperty, sValue); }
+
+    /// <summary>
+    /// The <see cref="ModelProperty" /> for the icon.
+    /// </summary>
+    static IntModelProperty IconProperty;
+
+    /// <summary>
+    /// Gets the icon to display.
+    /// </summary>
+    const MessageBoxIcon GetIcon() const { return static_cast<MessageBoxIcon>(GetValue(IconProperty)); }
+
+    /// <summary>
+    /// Sets the icon to display (default: None).
+    /// </summary>
+    void SetIcon(MessageBoxIcon nValue) { SetValue(IconProperty, static_cast<int>(nValue)); }
+
+    /// <summary>
+    /// The <see cref="ModelProperty" /> for the buttons to display.
+    /// </summary>
+    static IntModelProperty ButtonsProperty;
+
+    /// <summary>
+    /// Gets the buttons to display.
+    /// </summary>
+    const MessageBoxButtons GetButtons() const { return static_cast<MessageBoxButtons>(GetValue(ButtonsProperty)); }
+
+    /// <summary>
+    /// Sets the buttons to display (default: OK).
+    /// </summary>
+    void SetButtons(MessageBoxButtons nValue) { SetValue(ButtonsProperty, static_cast<int>(nValue)); }
+
+    /// <summary>
+    /// Shows a generic message.
+    /// </summary>
+    static void ShowMessage(const std::wstring& sMessage)
```

Comment:
Should have mentioned previously in the review. If you really want these to be explicitly static functions I'd suggest putting them in a helper class since static methods aren't member functions. 

The more "correct" way (at least according to Stroustrup) is to make them helper (non-class, non-static) functions in the same namespace the class it's supposed to help.

Reply:
Valid point. It's mostly a design thing but what you have is fine.